### PR TITLE
dmp-70 : Skeleton classes to Retrieve and Validate tokens

### PR DESCRIPTION
**Pull request by hemantasharma1129**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-70

### Change description ###

The commit contains the skeleton classes to implement Authentication process for DARTS.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/darts/config/security/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            );
+
+        return http.build();
+    }
+
+    @Bean
+    @SuppressWarnings({"PMD.SignatureDeclareThrowsException", "squid:S4502"})
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http
+            .csrf().disable()
```
[the remark this answers is not in the document]
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/reform/darts/config/security/SecurityConfig.java:46]: Do we need to fix this?

File: src/main/java/uk/gov/hmcts/reform/darts/controllers/authentication/AuthInternalUserController.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.ModelAndView;
+
+@Slf4j
+@RestController
+@RequestMapping("/internalUser")
+public class AuthInternalUserController extends AuthController {
+
+    @GetMapping("/login-or-refresh")
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/darts/controllers/authentication/AuthInternalUserController.java:14]: Do you think we should be consistent with the url names? For example, you have camel case for the request mapping part `/internalUser` and then kebab case for the get mapping part `/login-or-refresh`

File: src/test/java/uk/gov/hmcts/reform/darts/controllers/AuthInternalUserControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class AuthInternalUserControllerTest {
+
+    private final AuthInternalUserController authInternalUserController = new AuthInternalUserController();
+
+    @Test
+    void loginandrefreshtokenfromazureWhentokentoesntexistinsession() {
```
Comment (jackmaloney) [inline src/test/java/uk/gov/hmcts/reform/darts/controllers/AuthInternalUserControllerTest.java:16]: These test names are quite hard to read. I think the general pattern on HMCTS is to use camel case for test names

Comment (hemantasharma1129) [reply src/test/java/uk/gov/hmcts/reform/darts/controllers/AuthInternalUserControllerTest.java:16]: Well the ideal way is to use <methodName>_<expectedResult>_<condition> but the issue is Sonar criteria are bit messed up. I did use Camel casing but it only allowed me to use 2 upperCase test method name.

File: src/main/java/uk/gov/hmcts/reform/darts/controllers/authentication/AuthInternalUserController.java (hunk 2 above)
Comment (hemantasharma1129) [reply src/main/java/uk/gov/hmcts/reform/darts/controllers/authentication/AuthInternalUserController.java:14]: good point. I will update the mapping url to use underscore.

File: src/main/java/uk/gov/hmcts/reform/darts/config/security/SecurityConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (hemantasharma1129) [reply src/main/java/uk/gov/hmcts/reform/darts/config/security/SecurityConfig.java:46]: No we wont. Reasons below:
1. We are not using CSRF token. We will be using Azure JWT token for authentication.
2. Rest APis are stateless so no need of csrf.

**Pull request merged**